server/vulnerabilities/msrc/parsed/product.go

```diff
@@ -12,12 +15,53 @@ import (
 // (if any) and its version (if any).
 type Product string
 
+type Products map[string]Product
+
+var ErrNoMatch = errors.New("no product matches")
+
+func (p Products) GetMatchForOS(ctx context.Context, os fleet.OperatingSystem) (string, error) {
+	var dvMatch, noDvMatch string
+
+	for pID, p := range p {
+		normalizedOS := NewProductFromOS(os)
+		if p.Name() != normalizedOS.Name() {
+			continue
+		}
+
+		archMatch := p.Arch() == "all" || normalizedOS.Arch() == "all" || p.Arch() == normalizedOS.Arch()
+		if !archMatch {
+			continue
+		}
+
+		if p.HasDisplayVersion() && os.DisplayVersion != "" && strings.Index(string(p), os.DisplayVersion) != -1 {
+			dvMatch = pID
+			continue
+		}
+		if !p.HasDisplayVersion() {
```

Comment:
&& `os.DisplayVersion == ""`?

Reply:
it could be a bit risky, but we want to address the case where `os.DisplayVersion == "21H1"` matches even though it doesn't exist in the MSRC product list
```
"Products": {
        "11926": "Windows 11 for x64-based Systems",
        "11927": "Windows 11 for ARM64-based Systems",
        "12085": "Windows 11 Version 22H2 for ARM64-based Systems",
        "12086": "Windows 11 Version 22H2 for x64-based Systems",
        "12242": "Windows 11 Version 23H2 for ARM64-based Systems",
        "12243": "Windows 11 Version 23H2 for x64-based Systems"
    },
    ```

Reply:
Ah gotcha. Let's add a comment to clarify.